timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -228,6 +245,12 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                         width: this.stateController.canvasDisplayWidth
                     }
                     comp.update(opts);
+                    const cTop = comp.position.y;
+                    const cBottom = comp.height + cTop;
+                    console.log("Object " + cTop + " - " + cBottom + " / " + top + " - " + bottom);
+                    if (cTop <= bottom && top <= cBottom) {
```

Comment:
It appears that the vertical scrolling is not taken into account. The canvas bounds are unaffected by vertical scrolling, and the row y-position is fixed regardless of scrolling (e.g. it is the position in the full-height virtual scrollable area). Therefore it is always the first few rows of the model being considered visible, even when they are scrolled out of visible range.